tools/pipelines/templates/build-docker-service.yml

```diff
@@ -293,16 +293,41 @@ stages:
 
       # Pack
       - ${{ if eq(parameters.pack, true) }}:
+        # Need to create folders outside the docker container; if it creates them, the permissions will be wrong and
+        # we won't be able to write to them from outside the container.
+        - task: Bash@3
+          displayName: Create folder for pack artifact
+          inputs:
+            targetType: 'inline'
+            script: |
+              mkdir -p $(Build.ArtifactStagingDirectory)/pack
+
         - task: Docker@0
           displayName: Pack
           inputs:
             action: 'Run a Docker command'
-            customCommand: 'run -v $(System.DefaultWorkingDirectory)/pack/scoped:/usr/src/pack -t $(baseContainerTag) pnpm lerna exec --no-private -- mv `npm pack` /usr/src/pack'
+            customCommand: 'run -v $(Build.ArtifactStagingDirectory)/pack/scoped:/usr/src/pack -t $(baseContainerTag) pnpm lerna exec --no-private -- mv `npm pack` /usr/src/pack'
+
+        # Need to generate package lists manually, because we don't have access to the scripts/pack-packages.sh
+        # script that build-npm-package.yml uses.
+        - task: Bash@3
+          displayName: Generate package lists
+          env:
+            RELEASE_GROUP: ${{ parameters.tagName }}
+            STAGING_PATH: $(Build.ArtifactStagingDirectory)
+          inputs:
+            targetType: 'inline'
+            workingDirectory: ${{ parameters.buildDirectory }}
+            script: |
+              flub list --no-private --releaseGroup $RELEASE_GROUP --tarball --feed public > $STAGING_PATH/pack/packagePublishOrder-public.txt
+              flub list --no-private --releaseGroup $RELEASE_GROUP --tarball --feed internal-build > $STAGING_PATH/pack/packagePublishOrder-internal-build.txt
+              flub list --no-private --releaseGroup $RELEASE_GROUP --tarball --feed internal-dev > $STAGING_PATH/pack/packagePublishOrder-internal-dev.txt
+              flub list --no-private --releaseGroup $RELEASE_GROUP --tarball --feed internal-test > $STAGING_PATH/pack/packagePublishOrder-internal-test.txt
 
         - task: PublishPipelineArtifact@1
           displayName: Publish Artifact - pack
           inputs:
-            targetPath: '$(System.DefaultWorkingDirectory)/pack'
+            targetPath: '$(Build.ArtifactStagingDirectory)/pack'
```

Comment:
Updated everywhere where it relates to creating artifacts because I think it makes intentions clearer; that's where someone familiar with ADO pipelines would expect to see anything that is going to be published as artifacts later on.